### PR TITLE
fix(sdk): fixes outputfile key formatting

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
related #274

**Description of your changes:**
Fix a formatting bug with showing the bad output-file underscore format. Possibly related to #274.

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?
    
    If yes, use one of the following options:
  
    * **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update.
    *  After this PR is merged, create a cherry-pick PR to add these changes to the release branch. (For more information about creating a cherry-pick PR, see the [Kubeflow Pipelines release guide](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#option--git-cherry-pick).)

### CODE REVIEW

File: sdk/python/tests/compiler/compiler_tests.py [hunk 1]
```diff
@@ -184,8 +184,11 @@ def test_hidden_output_file_workflow(self):
     """
     Test compiling a workflow with non configurable output file.
     """
-    from .testdata.hidden_output_file import hidden_output_file_pipeline
-    self._test_pipeline_workflow(hidden_output_file_pipeline, 'hidden_output_file.yaml')
+    if sys.version_info < (3, 6, 0):
+      logging.warning("Skipping hidden_output workflow test for Python version < 3.6.0")
```
Comment (ckadner) [inline sdk/python/tests/compiler/compiler_tests.py:188]: could we add a one-line comment about why we require Python `3.6.1` or above?

Comment (Tomcli) [reply sdk/python/tests/compiler/compiler_tests.py:190]: Thanks @ckadner, I added a comment for it. It was due to some sorting function in the python 3.5 testing when we have certain combination of special characters.
https://travis-ci.com/github/kubeflow/kfp-tekton/jobs/384130977

**Review by ckadner — APPROVED** [review]
Thanks
/lgtm

**Pull request merged**